test/test-helpers.js

```diff
@@ -168,6 +168,26 @@ export const useFakeEnvironment = function(assert) {
   });
   fakeEnvironment.clock = sinon.useFakeTimers();
   fakeEnvironment.xhr = sinon.useFakeXMLHttpRequest();
+
+  // Sinon 1.10.2 handles abort incorrectly (triggering the error event)
+  // Later versions fixed this but broke the ability to set the response
+  // to an arbitrary object (in our case, a typed array).
+  XMLHttpRequest.prototype = Object.create(XMLHttpRequest.prototype);
+  XMLHttpRequest.prototype.abort = function abort() {
+    this.aborted = true;
+    this.response = this.responseText = '';
+    this.errorFlag = true;
+    this.requestHeaders = {};
```

Comment:
All this code comes from Sinon - the only thing I changed was to remove the dispatching of an `error` event at the end of the function.